### PR TITLE
add custom config option for EdgeCore config file

### PR DESCRIPTION
**Pull request by TianTianBigWang**
Signed-off-by: TianTianBigWang <zilong.wang@daocloud.io>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
use for `keadm beta join` command configurate the image repository from flag, and generate edgeCore config file
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by fisherxu — COMMENTED** [review]
I'm thinking whether we should add the config file to keadm? How do you think? @zhu733756 @zc2638

**Comment by zhu733756** [comment]
> I'm thinking whether we should add the config file to keadm? How do you think? @zhu733756 @zc2638

What kind of the configuration?  The current keadm already has the function of generating edgecore.yaml.

**Comment by zc2638** [comment]
> I'm thinking whether we should add the config file to keadm? How do you think? @zhu733756 @zc2638

The problem solved by this PR does not affect whether `keadm` needs a separate configuration file. 
@TianTianBigWang can open an issue to discuss this and refine it?

**Comment by TianTianBigWang** [comment]
> I'm thinking whether we should add the config file to keadm? How do you think? @zhu733756 @zc2638

This pr is only used to process the `edgeCore` configuration content after specifying a custom image. 

If we need to add a configuration file to `keadm`, we can discuss it later.

File: keadm/cmd/keadm/app/cmd/edge/joinbeta.go [hunk 1]
```diff
@@ -181,7 +181,11 @@ func createEdgeConfigFiles(opt *common.JoinOptions) error {
 	}
 	if edgeCoreConfig == nil {
 		klog.Infoln("The configuration does not exist or the parsing fails, and the default configuration is generated")
-		edgeCoreConfig = v1alpha1.NewDefaultEdgeCoreConfig()
+		cfgOpt := &v1alpha1.ConfigOptions{
+			ImageRepository: opt.ImageRepository,
+			KubeEdgeVersion: opt.KubeEdgeVersion,
+		}
+		edgeCoreConfig = v1alpha1.NewDefaultEdgeCoreConfig(cfgOpt)
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/edge/joinbeta.go:188]: Why not modify the `edgeCoreConfig` at here with `opt.ImageRepository` and `opt.KubeEdgeVersion` directly?

**Comment by fisherxu** [comment]
> > I'm thinking whether we should add the config file to keadm? How do you think? @zhu733756 @zc2638
> 
> The problem solved by this PR does not affect whether `keadm` needs a separate configuration file. @TianTianBigWang can open an issue to discuss this and refine it?

Sorry, I think I have made some mistakes :)

File: keadm/cmd/keadm/app/cmd/edge/joinbeta.go (hunk 1 above)
Comment (TianTianBigWang) [reply keadm/cmd/keadm/app/cmd/edge/joinbeta.go:188]: I think if we need to add other parameters to modify the configuration file in the future, it is more convenient to use a structure directly

Comment (fisherxu) [reply keadm/cmd/keadm/app/cmd/edge/joinbeta.go:188]: In futrue, if we need to add config field for edgecore, add it to edgecore's config directly. And if add the options for keadm, the config file for keadm is a better way, then converting the keadm config to edgecore. How do you think?

Comment (zhu733756) [reply keadm/cmd/keadm/app/cmd/edge/joinbeta.go:188]: > In futrue, if we need to add config field for edgecore, add it to edgecore's config directly. And if add the options for keadm, the config file for keadm is a better way, then converting the keadm config to edgecore. How do you think?

And also supports converting the edgecore.yaml to keadm.yaml is cool.

Comment (TianTianBigWang) [reply keadm/cmd/keadm/app/cmd/edge/joinbeta.go:188]: That is a good idea.
But I don't think it should affect this PR. 
For example, we might discuss the contents of the keadm configuration file, default parameters, etc. 
This should be a process.

**Pull request closed without merging**